datastore.it.ITDatastoreTest: testTransactionWithRead  and testTransactionWithQuery fail if the test DB type is Firestore in Datastore mode

[These tests](https://github.com/googleapis/java-datastore/blob/023229a/google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java#L240-L291) always timeout when testing with a Firestore in Datastore Mode DB. I think the reason they pass in continuous tests is because the type of DB they use (in project `gcloud-devel`) is "Cloud Datastore".
I believe the root cause of the timeouts is the transaction isolation and consistency behavior differences between the 2 Datastore DB types, documented [here](https://cloud.google.com/datastore/docs/concepts/transactions#transaction_locks) and [here](https://cloud.google.com/datastore/docs/concepts/transactions#transaction_locks). 
TL;DR 
With Firestore in datastore mode when 2 concurrent transactions read or write the same data, the lock held by one transaction can delay the other transaction (hence the timeouts), but with Cloud Datastore the contention is reported as an error at commit.


#### Environment details
All environments

#### Steps to reproduce

  1. [Create a Firestore in Datastore Mode DB](https://cloud.google.com/datastore/docs/quickstart#create_a_database) (or use an existing DB, if its type is known to be "Firestore in Datastore Mode")
  2. Set the GOOGLE_CLOUD_PROJECT env var to the project ID that owns the DB
  3. Set the GOOGLE_APPLICATION_CREDENTIALS env var to the absolute path of the key file for a service account that has permission to write to the DB
  4. run the tests, e.g. 
  ```
  GOOGLE_CLOUD_PROJECT=some-project GOOGLE_APPLICATION_CREDENTIALS=/tmp/sa-key.json mvn -P enable-integration-tests verify
  ```


#### Stack trace
**it.ITDatastoreTest.testTransactionWithQuery**
```
org.junit.runners.model.TestTimedOutException: test timed out after 100 seconds
	at java.net.SocketInputStream.socketRead0(Native Method)
	at java.net.SocketInputStream.socketRead(SocketInputStream.java:116)
	at java.net.SocketInputStream.read(SocketInputStream.java:171)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at sun.security.ssl.SSLSocketInputRecord.read(SSLSocketInputRecord.java:457)
	at sun.security.ssl.SSLSocketInputRecord.bytesInCompletePacket(SSLSocketInputRecord.java:68)
	at sun.security.ssl.SSLSocketImpl.readApplicationRecord(SSLSocketImpl.java:1332)
	at sun.security.ssl.SSLSocketImpl.access$300(SSLSocketImpl.java:73)
	at sun.security.ssl.SSLSocketImpl$AppInputStream.read(SSLSocketImpl.java:948)
	at java.io.BufferedInputStream.fill(BufferedInputStream.java:246)
	at java.io.BufferedInputStream.read1(BufferedInputStream.java:286)
	at java.io.BufferedInputStream.read(BufferedInputStream.java:345)
	at sun.net.www.http.HttpClient.parseHTTPHeader(HttpClient.java:735)
	at sun.net.www.http.HttpClient.parseHTTP(HttpClient.java:678)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream0(HttpURLConnection.java:1593)
	at sun.net.www.protocol.http.HttpURLConnection.getInputStream(HttpURLConnection.java:1498)
	at java.net.HttpURLConnection.getResponseCode(HttpURLConnection.java:480)
	at sun.net.www.protocol.https.HttpsURLConnectionImpl.getResponseCode(HttpsURLConnectionImpl.java:352)
	at com.google.api.client.http.javanet.NetHttpResponse.<init>(NetHttpResponse.java:36)
	at com.google.api.client.http.javanet.NetHttpRequest.execute(NetHttpRequest.java:149)
	at com.google.api.client.http.javanet.NetHttpRequest.execute(NetHttpRequest.java:84)
	at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1012)
	at com.google.datastore.v1.client.RemoteRpc.call(RemoteRpc.java:97)
	at com.google.datastore.v1.client.Datastore.commit(Datastore.java:85)
	at com.google.cloud.datastore.spi.v1.HttpDatastoreRpc.commit(HttpDatastoreRpc.java:162)
	at com.google.cloud.datastore.DatastoreImpl$5.call(DatastoreImpl.java:564)
	at com.google.cloud.datastore.DatastoreImpl$5.call(DatastoreImpl.java:561)
	at com.google.api.gax.retrying.DirectRetryingExecutor.submit(DirectRetryingExecutor.java:105)
	at com.google.cloud.RetryHelper.run(RetryHelper.java:76)
	at com.google.cloud.RetryHelper.runWithRetries(RetryHelper.java:50)
	at com.google.cloud.datastore.DatastoreImpl.commit(DatastoreImpl.java:560)
	at com.google.cloud.datastore.DatastoreImpl.commitMutation(DatastoreImpl.java:553)
	at com.google.cloud.datastore.DatastoreImpl.put(DatastoreImpl.java:513)
	at com.google.cloud.datastore.DatastoreHelper.put(DatastoreHelper.java:53)
	at com.google.cloud.datastore.DatastoreImpl.put(DatastoreImpl.java:488)
	at com.google.cloud.datastore.it.ITDatastoreTest.testTransactionWithQuery(ITDatastoreTest.java:284)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

**it.ITDatastoreTest.testTransactionWithRead**
this test would also timeout if [this code](https://github.com/googleapis/java-datastore/blob/023229a/google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java#L249-L251) was updating ENTITY3 as stated in the comment.
```
java.lang.AssertionError: Expecting a failure
	at org.junit.Assert.fail(Assert.java:89)
	at com.google.cloud.datastore.it.ITDatastoreTest.testTransactionWithRead(ITDatastoreTest.java:254)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

#### External references such as API reference guides
- https://cloud.google.com/datastore/docs/concepts/transactions#transaction_locks
- https://cloud.google.com/datastore/docs/upgrade-to-firestore#transactions

--- a/README.md
+++ b/README.md
@@ -50,7 +50,7 @@ If you are using Maven without the BOM, add this to your dependencies:
 If you are using Gradle 5.x or later, add this to your dependencies:
 
 ```Groovy
-implementation platform('com.google.cloud:libraries-bom:26.14.0')
+implementation platform('com.google.cloud:libraries-bom:26.15.0')
 
 implementation 'com.google.cloud:google-cloud-datastore'
 ```

--- a/google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java
+++ b/google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java
@@ -355,6 +355,8 @@ public class ITDatastoreTest {
 
     boolean t1AllPassed = statementExecutor.didAllPass("T1");
     boolean t2AllPassed = statementExecutor.didAllPass("T2");
+    // If two transactions conflict with each other, the database guarantees that only
+    // one can commit successfully at a time.
     // Using XOR to ensure that only one of transaction group is successful,
     boolean onlyOneTransactionIsSuccessful = t1AllPassed ^ t2AllPassed;
 
@@ -395,6 +397,8 @@ public class ITDatastoreTest {
 
     boolean t1AllPassed = statementExecutor.didAllPass("T1");
     boolean t2AllPassed = statementExecutor.didAllPass("T2");
+    // If two transactions conflict with each other, the database guarantees that only
+    // one can commit successfully at a time.
     // Using XOR to ensure that only one of transaction group is successful,
     boolean onlyOneTransactionIsSuccessful = t1AllPassed ^ t2AllPassed;
 

--- a/google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java
+++ b/google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java
@@ -29,6 +29,7 @@ import static org.junit.Assert.assertTrue;
 import static org.junit.Assert.fail;
 
 import com.google.cloud.Timestamp;
+import com.google.cloud.Tuple;
 import com.google.cloud.datastore.AggregationQuery;
 import com.google.cloud.datastore.Batch;
 import com.google.cloud.datastore.BooleanValue;
@@ -335,57 +336,69 @@ public class ITDatastoreTest {
   }
 
   @Test
-  public void testTransactionWithRead() {
-    Transaction transaction = DATASTORE.newTransaction();
-    assertNull(transaction.get(KEY3));
-    transaction.add(ENTITY3);
-    transaction.commit();
+  public void testTransactionWithRead() throws Exception{
+    StatementExecutor statementExecutor = new StatementExecutor();
+    Transaction baseTransaction = DATASTORE.newTransaction();
+    assertNull(baseTransaction.get(KEY3));
+    baseTransaction.add(ENTITY3);
+    baseTransaction.commit();
     assertEquals(ENTITY3, DATASTORE.get(KEY3));
 
-    transaction = DATASTORE.newTransaction();
-    assertEquals(ENTITY3, transaction.get(KEY3));
-    // update entity3 during the transaction
-    DATASTORE.put(Entity.newBuilder(ENTITY2).clear().set("from", "datastore").build());
-    transaction.update(Entity.newBuilder(ENTITY2).clear().set("from", "transaction").build());
-    try {
-      transaction.commit();
-      fail("Expecting a failure");
-    } catch (DatastoreException expected) {
-      assertEquals("ABORTED", expected.getReason());
-    }
+    Transaction transaction = DATASTORE.newTransaction();
+    statementExecutor.execute(
+        Tuple.of("T1", () -> assertEquals(ENTITY3, transaction.get(KEY3))),
+        // update entity3 during the transaction, will be blocked in case of pessimistic concurrency
+        Tuple.of("T2", () -> DATASTORE.put(Entity.newBuilder(ENTITY3).clear().set("from", "datastore").build())),
+        Tuple.of("T1", () -> transaction.update(Entity.newBuilder(ENTITY3).clear().set("from", "transaction").build())),
+        Tuple.of("T1", transaction::commit) // T1 will throw error in case of optimistic concurrency
+    );
+
+    boolean t1AllPassed = statementExecutor.didAllPass("T1");
+    boolean t2AllPassed = statementExecutor.didAllPass("T2");
+    // Using XOR to ensure that only one of transaction group is successful,
+    boolean onlyOneTransactionIsSuccessful = t1AllPassed ^ t2AllPassed;
+
+    assertThat(onlyOneTransactionIsSuccessful).isTrue();
   }
 
   @Test
-  public void testTransactionWithQuery() {
+  public void testTransactionWithQuery() throws Exception {
+    StatementExecutor statementExecutor = new StatementExecutor();
     Query<Entity> query =
         Query.newEntityQueryBuilder()
             .setKind(KIND2)
             .setFilter(PropertyFilter.hasAncestor(KEY2))
             .setNamespace(NAMESPACE)
             .build();
-    Transaction transaction = DATASTORE.newTransaction();
-    QueryResults<Entity> results = transaction.run(query);
-    assertTrue(results.hasNext());
-    assertEquals(ENTITY2, results.next());
-    assertFalse(results.hasNext());
-    transaction.add(ENTITY3);
-    transaction.commit();
+    Transaction baseTransaction = DATASTORE.newTransaction();
+    QueryResults<Entity> baseResults = baseTransaction.run(query);
+    assertTrue(baseResults.hasNext());
+    assertEquals(ENTITY2, baseResults.next());
+    assertFalse(baseResults.hasNext());
+    baseTransaction.add(ENTITY3);
+    baseTransaction.commit();
     assertEquals(ENTITY3, DATASTORE.get(KEY3));
 
-    transaction = DATASTORE.newTransaction();
-    results = transaction.run(query);
-    assertTrue(results.hasNext());
-    assertEquals(ENTITY2, results.next());
-    assertFalse(results.hasNext());
-    transaction.delete(ENTITY3.getKey());
-    // update entity2 during the transaction
-    DATASTORE.put(Entity.newBuilder(ENTITY2).clear().build());
-    try {
-      transaction.commit();
-      fail("Expecting a failure");
-    } catch (DatastoreException expected) {
-      assertEquals("ABORTED", expected.getReason());
-    }
+    Transaction transaction = DATASTORE.newTransaction();
+    statementExecutor.execute(
+        Tuple.of("T1", () -> {
+          QueryResults<Entity> results = transaction.run(query);
+          assertTrue(results.hasNext());
+          assertEquals(ENTITY2, results.next());
+          assertFalse(results.hasNext());
+        }),
+        Tuple.of("T1", () -> transaction.delete(ENTITY3.getKey())),
+        // update entity2 during the transaction, will be blocked in case of pessimistic concurrency
+        Tuple.of("T2", () -> DATASTORE.put(Entity.newBuilder(ENTITY2).clear().build())),
+        Tuple.of("T1", transaction::commit) // T1 will throw error in case of optimistic concurrency
+    );
+
+    boolean t1AllPassed = statementExecutor.didAllPass("T1");
+    boolean t2AllPassed = statementExecutor.didAllPass("T2");
+    // Using XOR to ensure that only one of transaction group is successful,
+    boolean onlyOneTransactionIsSuccessful = t1AllPassed ^ t2AllPassed;
+
+    assertThat(onlyOneTransactionIsSuccessful).isTrue();
   }
 
   @Test

--- a/google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java
+++ b/google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java
@@ -336,7 +336,7 @@ public class ITDatastoreTest {
   }
 
   @Test
-  public void testTransactionWithRead() throws Exception{
+  public void testTransactionWithRead() throws Exception {
     StatementExecutor statementExecutor = new StatementExecutor();
     Transaction baseTransaction = DATASTORE.newTransaction();
     assertNull(baseTransaction.get(KEY3));
@@ -348,10 +348,17 @@ public class ITDatastoreTest {
     statementExecutor.execute(
         Tuple.of("T1", () -> assertEquals(ENTITY3, transaction.get(KEY3))),
         // update entity3 during the transaction, will be blocked in case of pessimistic concurrency
-        Tuple.of("T2", () -> DATASTORE.put(Entity.newBuilder(ENTITY3).clear().set("from", "datastore").build())),
-        Tuple.of("T1", () -> transaction.update(Entity.newBuilder(ENTITY3).clear().set("from", "transaction").build())),
+        Tuple.of(
+            "T2",
+            () ->
+                DATASTORE.put(Entity.newBuilder(ENTITY3).clear().set("from", "datastore").build())),
+        Tuple.of(
+            "T1",
+            () ->
+                transaction.update(
+                    Entity.newBuilder(ENTITY3).clear().set("from", "transaction").build())),
         Tuple.of("T1", transaction::commit) // T1 will throw error in case of optimistic concurrency
-    );
+        );
 
     boolean t1AllPassed = statementExecutor.didAllPass("T1");
     boolean t2AllPassed = statementExecutor.didAllPass("T2");
@@ -383,17 +390,19 @@ public class ITDatastoreTest {
 
     Transaction transaction = DATASTORE.newTransaction();
     statementExecutor.execute(
-        Tuple.of("T1", () -> {
-          QueryResults<Entity> results = transaction.run(query);
-          assertTrue(results.hasNext());
-          assertEquals(ENTITY2, results.next());
-          assertFalse(results.hasNext());
-        }),
+        Tuple.of(
+            "T1",
+            () -> {
+              QueryResults<Entity> results = transaction.run(query);
+              assertTrue(results.hasNext());
+              assertEquals(ENTITY2, results.next());
+              assertFalse(results.hasNext());
+            }),
         Tuple.of("T1", () -> transaction.delete(ENTITY3.getKey())),
         // update entity2 during the transaction, will be blocked in case of pessimistic concurrency
         Tuple.of("T2", () -> DATASTORE.put(Entity.newBuilder(ENTITY2).clear().build())),
         Tuple.of("T1", transaction::commit) // T1 will throw error in case of optimistic concurrency
-    );
+        );
 
     boolean t1AllPassed = statementExecutor.didAllPass("T1");
     boolean t2AllPassed = statementExecutor.didAllPass("T2");

--- a/google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java
+++ b/google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/ITDatastoreTest.java
@@ -356,7 +356,7 @@ public class ITDatastoreTest {
     boolean t1AllPassed = statementExecutor.didAllPass("T1");
     boolean t2AllPassed = statementExecutor.didAllPass("T2");
     // If two transactions conflict with each other, the database guarantees that only
-    // one can commit successfully at a time.
+    // one can commit successfully at a time. Please refer to StatementExecutor class for more info.
     // Using XOR to ensure that only one of transaction group is successful,
     boolean onlyOneTransactionIsSuccessful = t1AllPassed ^ t2AllPassed;
 
@@ -398,7 +398,7 @@ public class ITDatastoreTest {
     boolean t1AllPassed = statementExecutor.didAllPass("T1");
     boolean t2AllPassed = statementExecutor.didAllPass("T2");
     // If two transactions conflict with each other, the database guarantees that only
-    // one can commit successfully at a time.
+    // one can commit successfully at a time. Please refer to StatementExecutor class for more info.
     // Using XOR to ensure that only one of transaction group is successful,
     boolean onlyOneTransactionIsSuccessful = t1AllPassed ^ t2AllPassed;
 

--- a/google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/StatementExecutor.java
+++ b/google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/StatementExecutor.java
@@ -60,8 +60,8 @@ class StatementExecutor {
         // waiting for statement to execute
         future.get(10, SECONDS);
       } catch (Exception exception) {
+        future.cancel(true);
         if (transactionConflict(exception)) {
-          future.cancel(true);
           failures.put(groupId, exception);
         } else {
           throw exception;
@@ -78,12 +78,12 @@ class StatementExecutor {
   private boolean transactionConflict(Exception exception) {
     if (exception instanceof TimeoutException) { // timed out coz of pessimistic concurrency delay
       return true;
-    } else {
-      return exception instanceof ExecutionException
-          && exception.getCause().getClass() == DatastoreException.class
-          && exception.getMessage()
-          .contains("contention"); // exception raise coz of optimistic concurrency
     }
+    return exception instanceof ExecutionException
+        && exception.getCause().getClass() == DatastoreException.class
+        && exception.getMessage()
+        .contains("contention"); // exception raise coz of optimistic concurrency
+
   }
 
   interface Statement {

--- a/google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/StatementExecutor.java
+++ b/google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/StatementExecutor.java
@@ -1,0 +1,93 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.datastore.it;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import com.google.cloud.Tuple;
+import com.google.cloud.datastore.DatastoreException;
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.Multimap;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeoutException;
+
+
+/**
+ * An executor class to handle interleaved transactions.
+ *
+ * It executes statements (under multiple transactions) and record their failures under a groupId
+ * provided by users.
+ */
+class StatementExecutor {
+
+  private final Multimap<String, Exception> failures = ArrayListMultimap.create();
+
+  /**
+   * Executes a list of {@link Statement} one by one and record their failures under the groupId. In
+   * case of pessimistic concurrency, a statement will be blocked and cause delay until another
+   * transaction which was started earlier is committed. In case of optimistic concurrency, both
+   * transaction can perform their operation simultaneously, but the one which commits first will be
+   * a winner and other one will get an error on commit operation indicating a need for retry.
+   *
+   * @param tuples A {@link Statement(String, String) Tuple(&lt;String, Statement&gt;)} has a
+   *     groupId of {@link String} type and a {@link Statement} to execute.
+   */
+  @SafeVarargs
+  final void execute(Tuple<String, Statement>... tuples) throws Exception {
+    ExecutorService executorService = Executors.newSingleThreadExecutor();
+    for (Tuple<String, Statement> tuple : tuples) {
+      String groupId = tuple.x();
+      Statement statement = tuple.y();
+      Future<?> future = executorService.submit(statement::execute);
+      try {
+        // waiting for statement to execute
+        future.get(10, SECONDS);
+      } catch (Exception exception) {
+        if (transactionConflict(exception)) {
+          future.cancel(true);
+          failures.put(groupId, exception);
+        } else {
+          throw exception;
+        }
+      }
+    }
+    executorService.shutdown();
+  }
+
+  boolean didAllPass(String groupId) {
+    return failures.get(groupId).isEmpty();
+  }
+
+  private boolean transactionConflict(Exception exception) {
+    if (exception instanceof TimeoutException) { // timed out coz of pessimistic concurrency delay
+      return true;
+    } else {
+      return exception instanceof ExecutionException
+          && exception.getCause().getClass() == DatastoreException.class
+          && exception.getMessage()
+          .contains("contention"); // exception raise coz of optimistic concurrency
+    }
+  }
+
+  interface Statement {
+    void execute();
+  }
+
+}

--- a/google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/StatementExecutor.java
+++ b/google-cloud-datastore/src/test/java/com/google/cloud/datastore/it/StatementExecutor.java
@@ -28,11 +28,10 @@ import java.util.concurrent.Executors;
 import java.util.concurrent.Future;
 import java.util.concurrent.TimeoutException;
 
-
 /**
  * An executor class to handle interleaved transactions.
  *
- * It executes statements (under multiple transactions) and record their failures under a groupId
+ * <p>It executes statements (under multiple transactions) and record their failures under a groupId
  * provided by users.
  */
 class StatementExecutor {
@@ -81,13 +80,12 @@ class StatementExecutor {
     }
     return exception instanceof ExecutionException
         && exception.getCause().getClass() == DatastoreException.class
-        && exception.getMessage()
-        .contains("contention"); // exception raise coz of optimistic concurrency
-
+        && exception
+            .getMessage()
+            .contains("contention"); // exception raise coz of optimistic concurrency
   }
 
   interface Statement {
     void execute();
   }
-
 }

